sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CollationTypeCasts.scala

```diff
@@ -152,6 +150,8 @@ object CollationTypeCasts extends TypeCoercionRule {
       case 0 =>
         val implicitTypes = expr.filter {
             case Literal(_, _: StringType) => false
+            case cast: Cast if cast.getTagValue(Cast.USER_SPECIFIED_CAST).isDefined =>
```

Comment:
why do we need this change? The default case match returns true here.

Reply:
Good catch, removed it.